src/SourceTerms/CPU_SrcSolver.cpp

```diff
@@ -59,6 +59,12 @@ void CPU_SrcSolver( const real h_Flu_Array_In [][FLU_NIN_S ][ CUBE(SRC_NXT)
 #  endif
 
 
+// EoS is not defined when MODEL != HYDRO
+#  if ( MODEL != HYDRO )
+   EoS_t EoS;
+#  endif
```

Comment:
The current strategy in GAMER is to exclude global variables from `Main.cpp` if they are not relevant to the adopted configuration (e.g., ELBDM-specific variables are defined only when `MODEL == ELBDM`). This approach aims to prevent accidental misuse of these variables. When needed, they are defined locally instead (for instance, see [this example](https://github.com/gamer-project/gamer/blob/main/src/Main/InvokeSolver.cpp#L460)).

I agree that this approach is not ideal and can be confusing, so I’m open to exploring alternative solutions. However, I suggest addressing this in a separate PR, where we can update all global variables consistently.

What are your thoughts? Please feel free to file a feature request for the record.

Reply:
I see. I do not have a better solution now.

> Please feel free to file a feature request for the record.

How about recording this conversation to #299?

Reply:
Agree.